### ISSUE TITLE
Phase masks need work after new paramater initialization logic

### ISSUE THREAD

**Issue by diptodip** [TuragaLab/chromatix#12]
In the translation to the new parameter initialization logic, we introduced a few bugs into the phase mask elements, especially `SpatialLightModulator`.

Specifically:

* both elements should call `phase_change`, and `SpatialLightModulator` shouldn't call `jnp.exp` twice on the phase!
* `SpatialLightModulator` should only call `wrap_phase()` once
* we should change the arguments and logic for the phase mask initialization functions so that they don't require a `Field`, since we don't intend for them to be called in that way anymore
* which means that the phase mask initialization functions should just accept as arguments only what they need from: `shape, spacing, wavelength, n, f, NA` and then construct a grid as needed
